src/machine.rs

```diff
@@ -420,6 +420,34 @@ impl<'tcx> PrimitiveLayouts<'tcx> {
     }
 }
 
+/// A thread's CPU affinity mask determines the set of CPUs on which it is eligible to run.
+#[derive(Clone)]
+pub(crate) struct CpuAffinityMask(pub(crate) [u8; 128]);
+
+impl CpuAffinityMask {
+    // the default affinity mask includes only the available CPUs
+    pub fn new(mut cpu_count: u32) -> Self {
+        let mut bits = [0; 128];
+
+        // set the first cpu_count bits. The format is as follows:
+        //
+        // cpu 0    : [ 0b0000_0001, 0, 0, ..., 0 ]
+        // cpu 0, 1 : [ 0b0000_0011, 0, 0, ..., 0 ]
+        // cpu 0, 8 : [ 0b0000_0001, 0b0000_0001, 0, ..., 0 ]
+        for byte in bits.iter_mut() {
+            // set the lowest N bits of a byte
+            *byte = (0xFFu16 << Ord::min(8, cpu_count) >> 8) as u8;
```

Comment:
Please add parentheses, I can't even parse this.

What's N here? cpu_count? No I guess it must be `Ord::min(8, cpu_count)`? Maybe let-bind that first.